### PR TITLE
Feature 1697 determine server https

### PR DESCRIPTION
**Pull request by taojing2002**

### CODE REVIEW

File: src/edu/ucsb/nceas/metacat/DocumentImpl.java
Comment (artntek) [inline src/edu/ucsb/nceas/metacat/DocumentImpl.java:1]: Can we take care of these while touching this file?

• 'Integer(java.lang.String)' is deprecated since version 9 and marked for removal :248
• 'Integer(java.lang.String)' is deprecated since version 9 and marked for removal :404
• 'Integer(int)' is deprecated since version 9 and marked for removal :627
• 'Integer(int)' is deprecated since version 9 and marked for removal :1339
• 'Integer(int)' is deprecated since version 9 and marked for removal :1476
• 'Integer (java.lang. String)' is deprecated since version 9 and marked for removal :1539
• 'Integer(java.lang.String)' is deprecated since version 9 and marked for removal: 1595
• 'Integer(int)' is deprecated since version 9 and marked for removal: 1730
• 'Integer(int)' is deprecated since version 9 and marked for removal: 1945
• 'Integer(java.lang.String)' is deprecated since version 9 and marked for removal

File: src/edu/ucsb/nceas/metacat/doi/ezid/EzidDOIService.java
Comment (artntek) [inline src/edu/ucsb/nceas/metacat/doi/ezid/EzidDOIService.java:1]: Can we take care of these while touching this file?

'Integer(int)' is deprecated since version 9 and marked for removal :354
'Integer(int)' is deprecated since version 9 and marked for removal :361

**Review by artntek — CHANGES_REQUESTED** [review]

File: build.xml
[the remark this answers is not in the document]
Comment (taojing2002) [reply build.xml:1]: I got your point. However, the reason I deleted the package is one of its class used the properties which are removed in this branch. Rather than modifying code, I decided to remove the whole package. So I guess it is reasonable to keep the change in this PR. What do you think?

[the remark this answers is not in the document]
Comment (artntek) [reply build.xml:1]: That's fine - not a big deal - just a suggestion for next time. An example workaround would be to delete the package first, as one PR, and then delete the properties in the second PR, so you don't break a dependency.

No problem

File: src/edu/ucsb/nceas/metacat/admin/PropertiesAdmin.java [hunk 1]
```diff
@@ -98,16 +98,15 @@ public void configureProperties(HttpServletRequest request,
...
-                if (isNotSet(serverHttpSSLPort)) {
-                    PropertyService.setPropertyNoPersist(serverHttpSSLPort,
+                final String serverPort = "server.port";
+                if (isNotSet(serverPort)) {
+                    PropertyService.setPropertyNoPersist(serverPort,
                             SystemUtil.discoverServerSSLPort(request));
                 }
+                final String serverHTTPS = "server.https";
+                if (isNotSet(serverHTTPS)) {
+                    PropertyService.setPropertyNoPersist(serverHTTPS, "true");
```
[the remark this answers is not in the document]
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/admin/PropertiesAdmin.java:108]: ok - makes sense

File: src/edu/ucsb/nceas/metacat/util/SystemUtil.java [hunk 2]
```diff
@@ -125,55 +125,36 @@ public static String discoverServerSSLPort(HttpServletRequest request) {
...
-            serverURL = "https://";
-        }
-        
-        serverURL += PropertyService.getProperty("server.name");
-        
-        if (!httpPort.equals("80") && !httpPort.equals("443")) {
-            serverURL += ":" + httpPort;
+        String serverURL = "https://";
+        String serverHttps = PropertyService.getProperty("server.https");
+        if (serverHttps != null && serverHttps.equalsIgnoreCase("false")) {
```
[the remark this answers is not in the document]
Comment (taojing2002) [reply src/edu/ucsb/nceas/metacat/util/SystemUtil.java:136]: Thank you for the information! This is my bad habit to check for `null` before comparing strings :). I think this is a defensive way - it wouldn't hurt anything but if the behavior of `PropertyService.getProperty()` changes, our code wouldn't throw a null exception. What do you think?

[the remark this answers is not in the document]
Comment (artntek) [reply src/edu/ucsb/nceas/metacat/util/SystemUtil.java:136]: > This is my bad habit to check for null before comparing strings :).

Don't get me wrong - I think this is a very good habit. However, in cases where an API makes a guarantee (e.g. [getProperty() will never return null](https://github.com/NCEAS/metacat/blob/1f770e5dbb8dcd8b028289c3dc9af10402b6a4a9/src/edu/ucsb/nceas/metacat/properties/PropertyService.java#L164)), then I think it is better to remove the null check. That makes the code simpler and easier to read, and it also maintains consistency (i.e. most places in the code do not have a null check when calling `getProperty()`). Otherwise, someone new to the code wonders why there's a null check (when the method never returns null), and this leads to cognitive dissonance, confusion (feelings of "did I miss something important??") and therefore wasted time).

I hear your concern about the possibility of getProperty() starting to return `null` in future, but that is extremely unlikely. If it does change, it will be a huge refactor anyway 😬. 

In summary, I think it's generally better to have consistent, readable, understandable code, instead of adding contingencies for a possible future change that is very unlikely to happen.

[the remark this answers is not in the document]
Comment (mbjones) [reply src/edu/ucsb/nceas/metacat/util/SystemUtil.java]: Good points all around. I also note that wrapping the null check up with the empty string check could both make it more readable when we need to do both, and standardize the check. Apache common-lang even has some `StringUtils` methods for this common need, although they are so simple it would likely be better to just implement them in our software directly.

`StringUtils.isEmpty(String str)` - Checks if a String is null or empty ("")
`StringUtils.isBlank(String str)` - Checks if a String is null, whitespace (like " "), or empty ("")
